xarray/coding/variables.py

```diff
@@ -93,7 +93,7 @@ def lazy_elemwise_func(array, func, dtype):
     -------
     Either a dask.array.Array or _ElementwiseFunctionArray.
     """
-    if isinstance(array, dask_array_type):
+    if is_duck_dask_array(array):
         return array.map_blocks(func, dtype=dtype)
```

Comment:
does every `duck dask array` define `map_blocks`, or should we use `dask.array.map_blocks(func, array, ...)` instead?

Reply:
I *think* that methods like this are defined by `duck dask arrays` and I say that only based on `rechunk` being defined by them.

Reply:
It's probably safer to use `dask.array.map_blocks`

Reply:
if we do that, it might be good to change all other instances of `array.map_blocks(func, ...)` to `map_blocks(func, array, ...)`, too. I think there are only 4 calls: in `xarray.core.dask_array_ops` and `xarray.core.dask_array_compat`, with different names for `array`.

This is not really related to this PR though, so you could also leave that change to a new PR.

Reply:
:+1: